File: be/app/models/weather.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime


class Location(BaseModel):
    city: str
    country: Optional[str] = None
    lat: Optional[float] = None
    lon: Optional[float] = None


class WeatherData(BaseModel):
    location: Location
    timestamp: datetime
    current: dict
    forecast: Optional[List[dict]] = None


class SavedLocation(BaseModel):
    user_id: Optional[str] = None
    location: Location
    is_default: bool = False

File: fe/src/types/weather.ts
export interface WeatherData {
  city: string,
  country?: string,
  temperature: number,
  feels_like?: number,
  humidity: number,
  pressure: number,
  wind_speed: number,
  description: string,
  icon?: string,
  timestamp: string,
  coordinates?: {
    lat: number,
    lon: number
  }
}

export interface ForecastItem {
  date: string;
  time: string;
  temperature: number;
  humidity: number;
  pressure: number;
  wind_speed: number;
  description: string;
  icon?: string;
}

export interface ForecastData {
  city: string;
  country?: string;
  list: ForecastItem[];
}

export interface SavedLocation {
  id?: string;
  user_id?: string;
  location: {
    city: string;
    country?: string;
    lat?: number;
    lon?: number;
  };
  is_default: boolean;
}
